Update host documentation for versions v1.3 and v1.4 to add a note on… (#745)
harvester/docs: docs/host/host.md

  When you remove a control plane node, a worker node will be promoted to control plane node. Harvester v1.3.0 allows you to assign a role to each node that joins a cluster. In earlier Harvester versions, worker nodes were randomly selected for promotion. If you prefer to promote specific nodes, please see [Role Management](./host.md#role-management) and [Harvester Configuration](../install/harvester-configuration.md#installrole) for more information.

  :::note

>   Harvester does not automatically promote a worker node when a control plane node becomes unavailable for any reason. Promotion occurs only when a control plane node is removed from the cluster.

Reviewer: So this is saying that promotion only occurs if the control plane node is removed (detached) from the cluster. It is not a complete control plane health check mechanism to guarantee high availability, right?

Can we say something like:

"The auto node promotion mechanism occurs only when a control plane node "unjoin" (not sure if this is a word) a cluster. It doesn't happen if the node is unavailable due to failed health checks."

@jillian-maroket WDYT?

Reviewer 2: @ihcsim Got it. Here's the reworded paragraph:

_Automatic node promotion occurs only when a control plane node detaches itself from the cluster. This does not include situations wherein a node becomes unavailable due to failed health checks._

>   Automatic node promotion occurs only when a control plane node detaches itself from the cluster. This does not include situations wherein a node becomes unavailable due to failed health checks.

Reviewer: ```suggestion
  Automatic node promotion occurs only when a control plane node is deleted from the cluster. This does not include situations wherein a node becomes unavailable due to failed health checks. The unhealthy node retains its role.
```

  When you remove a control plane node, a worker node will be promoted to control plane node. Harvester v1.3.0 allows you to assign a role to each node that joins a cluster. In earlier Harvester versions, worker nodes were randomly selected for promotion. If you prefer to promote specific nodes, please see [Role Management](./host.md#role-management) and [Harvester Configuration](../install/harvester-configuration.md#installrole) for more information.

> :::note
> 
> Harvester does not automatically promote a worker node to a control plane node if a control plane node goes offline or fails due to some reason, promotion only occurs when a control plane node is removed from the cluster.
> 
> :::

Reviewer 2: ```suggestion
  :::note

  Harvester does not automatically promote a worker node when a control plane node becomes unavailable for any reason. Promotion occurs only when a control plane node is removed from the cluster.

  :::
```
